src/ucp/core/ucp_context.c

```diff
@@ -486,22 +486,33 @@ static int ucp_tls_array_is_present(const char **tls, unsigned count,
     }
 }
 
-static int ucp_config_is_tl_enabled(const char **names, unsigned count,
+static int ucp_config_is_tl_enabled(const ucs_config_allow_list_t *allow_list,
                                     const char *tl_name, int is_alias,
                                     uint8_t *rsc_flags, uint64_t *tl_cfg_mask)
 {
     char strict_name[UCT_TL_NAME_MAX + 1];
+    int is_in_list;
 
     snprintf(strict_name, sizeof(strict_name), "\\%s", tl_name);
-    return /* strict name, with leading \\ */
-           (!is_alias && ucp_tls_array_is_present(names, count, strict_name, "",
-                                                  rsc_flags, tl_cfg_mask)) ||
-           /* plain transport name */
-           ucp_tls_array_is_present(names, count, tl_name, "", rsc_flags,
-                                    tl_cfg_mask) ||
-           /* all available transports */
-           ucp_tls_array_is_present(names, count, UCP_RSC_CONFIG_ALL, "", rsc_flags,
-                                    tl_cfg_mask);
+
+    is_in_list = /* strict name, with leading \\ */
+            (!is_alias &&
+             (ucp_tls_array_is_present((const char**)allow_list->array.names,
+                                       allow_list->array.count, strict_name, "",
+                                       rsc_flags, tl_cfg_mask))) ||
+            /* plain transport name */
+            (ucp_tls_array_is_present((const char**)allow_list->array.names,
+                                      allow_list->array.count, tl_name, "",
+                                      rsc_flags, tl_cfg_mask));
+
+    /* The tl is considered "enabled" if:
+       (1) it is in the list, and the list is in "allow" mode, or 
+       (2) it is not in the list, when the list is in "negate" mode */
+    if (allow_list->mode == UCS_CONFIG_ALLOW_LIST_ALLOW) {
+        return is_in_list;
+    } else {
+        return !is_in_list;
+    }
```

Comment:
maybe better remove whether tl_name is found in names? Seems this would simplify  further processing in `ucp_is_resource_in_transports_list`

Reply:
@brminich , What do you mean? Remove this function and inline it in `ucp_is_resource_in_transports_list`? It is used there twice..

Reply:
No, I meant to leave this function, but make it return whether the name is present  `allow_list->array.names`. Also would need to rename it, like `ucp_config_is_tl_name_present`. 
Then it would be easier to handle its results in `ucp_is_resource_in_transports_list`